File: Chapter 05/Python/section_5_4_server_monitor/after/infrastructure/adapters/twilio_adapter.py
from core.ports.alert_port import AlertPort
from infrastructure.external_libs.fake_libs import TwilioClient

class TwilioAdapter(AlertPort):
    """
    ADAPTER 1: The 'Real' Production Adapter.
    This class is the bridge between the internal AlertPort and the external Twilio API.
    """

    def __init__(self, api_key: str, target_phone_number: str):
        """Configuration is injected here, keeping 'God Mode' keys out of the Core."""
        self.api_key = api_key
        self.target_phone_number = target_phone_number

    def send_alert(self, message: str) -> None:
        # We encapsulate the 'Chaotic' 3rd party SDK here.
        client = TwilioClient(self.api_key)
        client.send_sms(self.target_phone_number, message)
        print(f"(PROD ADAPTER) SMS sent via Twilio: {message}")

File: Chapter 05/JS/section_5_4_server_monitor/after/infrastructure/adapters/twilioAdapter.js
const AlertPort = require('../../core/ports/alertPort');
const { TwilioClient } = require('../externalLibs/fakeLibs');

/**
 * ADAPTER 1: The "Real" Production Adapter.
 * This class is the bridge between the internal AlertPort and the external Twilio API.
 */
class TwilioAdapter extends AlertPort {
    /**
     * Configuration is injected here, keeping 'God Mode' keys out of the Core.
     */
    constructor(apiKey, targetPhoneNumber) {
        super();
        this.apiKey = apiKey;
        this.targetPhoneNumber = targetPhoneNumber;
    }

    sendAlert(message) {
        // We encapsulate the 'Chaotic' 3rd party SDK here.
        const client = new TwilioClient(this.apiKey);
        client.sendSms(this.targetPhoneNumber, message);
        console.log(`(PROD ADAPTER) SMS sent via Twilio: ${message}`);
    }
}

module.exports = TwilioAdapter;